ts/features/messages/hooks/usePNOptInMessage.ts

```diff
@@ -0,0 +1,26 @@
+import { isPNOptInMessage } from "../../pn/utils";
+import { useOnFirstRender } from "../../../utils/hooks/useOnFirstRender";
+import {
+  trackPNOptInMessageCTADisplaySuccess,
+  trackPNOptInMessageOpened
+} from "../../pn/analytics";
+import { ServiceId } from "../../../../definitions/backend/ServiceId";
+import { useIOStore } from "../../../store/hooks";
+
+export const usePNOptInMessage = (serviceId: ServiceId) => {
+  const store = useIOStore();
```

Comment:
Why do we need to directly use the store here instead of a standard selector? Do we expect the output of `isPNOptInMessage` to change while a message is displayed and cause an un-neccessary re-rendering of the component?